### PR TITLE
Adding initial verifier example for Circom

### PR DESCRIPTION
**Pull request by katiemckeon**
This PR will introduce an end-to-end example of uploading a Circom circuit and producing a smart verifier contract from that circuit to our `reference_code/` directory.  There are a few non-trivial steps turning the fields we return from the API into data structures interpretable by solidity.  The code in this branch automates those steps.

## Reviewer Checklist
- [x] follow along with `reference_code/verifiers/circom/solidity/README.md`
  - [x] Was any necessary step not listed in the README?
  - [x] Do you see a successful test of the smart contract in the end? (It will look like this:
```
Stdout: Compiled 1 Solidity file successfully (evm target: paris).


  Groth16Verifier
    ✔ Should verify a valid proof (2213ms)


  1 passing (2s)
```

### CODE REVIEW

File: reference_code/verifiers/circom/README.md [hunk 1]
```diff
@@ -0,0 +1,3 @@
+## Circom Verifiers
```
Comment (katiemckeon) [inline reference_code/verifiers/circom/README.md:1]: Note that I'm nesting two layers here: `verifiers/circom/solidity`.  I think the eventual structure we want to have is `verifiers/{framework}/{verifier_framework}`.

File: reference_code/verifiers/circom/solidity/.gitignore [hunk 2]
```diff
@@ -0,0 +1,15 @@
+/build
```
Comment (KPreisner) [inline reference_code/verifiers/circom/solidity/.gitignore:1]: .gitignore this file

Comment (katiemckeon) [reply reference_code/verifiers/circom/solidity/.gitignore:1]: addressed by https://github.com/Sindri-Labs/sindri-resources/pull/68/commits/46547a7e99eefaeb2f12f563d26072efc2121751

**Review by stasiacarson — APPROVED** [review]
LFGTM

**Pull request merged**